akka-persistence/src/main/scala/akka/persistence/state/javadsl/DurableStateUpdateStore.scala

```diff
@@ -20,5 +20,9 @@ trait DurableStateUpdateStore[A] extends DurableStateStore[A] {
    */
   def upsertObject(persistenceId: String, revision: Long, value: A, tag: String): CompletionStage[Done]
 
-  def deleteObject(persistenceId: String): CompletionStage[Done]
+  @deprecated(message = "Use the deleteObject overload with revision instead.", since = "2.6.20")
+  def deleteObject(@nowarn persistenceId: String): CompletionStage[Done] = CompletableFuture.completedFuture(Done)
+
+  @nowarn
+  def deleteObject(persistenceId: String, @nowarn revision: Long): CompletionStage[Done] = deleteObject(persistenceId)
```

Comment:
I think we should keep both methods without implementations. Better to have it as a "pure" interface without default (wrong) implementations in the future. Since `deleteObject` was never called from Akka previously I don't think we should care too much. If a plugin implements this and update to latest Akka they will have to implement both methods. That is not great, but better than "wrong" implementations that plugin authors don't notice.

Reply:
In that case, the `DurableStateStoreInteractions` will have to call both overloads and expect only one to be implemented. I could also see implementations throwing `NotImplementedError` using `???`.

Reply:
Or using `flatMap` / `map` to order the calls:
```
  protected def internalDelete(
      ctx: ActorContext[InternalProtocol],
      cmd: Any,
      state: Running.RunningState[S]): Running.RunningState[S] = {

    val newRunningState = state.nextRevision().copy(state = setup.emptyState)
    val persistenceId = setup.persistenceId.id

    onDeleteInitiated(ctx, cmd)

    implicit val ec = ctx.executionContext

    @nowarn
    val f = for {
      _ <- setup.durableStateStore.deleteObject(persistenceId).recover {
        case _: NotImplementedError => Done
      }

      _ <- setup.durableStateStore.deleteObject(persistenceId, newRunningState.revision).recover {
        case _: NotImplementedError => Done
      }
    } yield Done

    ctx.pipeToSelf[Done](f) {
      case Success(_)     => InternalProtocol.DeleteSuccess
      case Failure(cause) => InternalProtocol.DeleteFailure(cause)
    }

    newRunningState
  }
  ```

Reply:
Thanks for giving it more thoughts, but let's not complicate this more than necessary. The method without revision is dead, we should not call it. We just keep it deprecated for binary compatibility, but it would only be used if someone used the store directly without using Akka's DurableStateBehavior.

It's ok that DurableStateStoreInteractions only calls the new method.